Makefile

```diff
@@ -52,6 +52,13 @@ ifeq ($(LIBOPENSSL),0)
 	override LIB_DEPENDS += openssl
 endif
 
+GLIBC_MAJOR_VERSION := $(shell ldd --version | awk '/ldd/{print $NF}' | cut -f1 -d.)
+GLIBC_MINOR_VERSION := $(shell ldd --version | awk '/ldd/{print $NF}' | cut -f2 -d.)
+GLIBC_GE_2_25 := $(shell [ $(GLIBC_MAJOR_VERSION) -gt 2 -o \( $(GLIBC_MAJOR_VERSION) -eq 2 -a $(GLIBC_MINOR_VERSION) -ge 25 \) ] && echo true)
+ifeq ($(GLIBC_GE_2_25),true)
+	CFLAGS += -DHAVE_SYS_RANDOM
```

Comment:
Sorry to pester you again, but wouldn't it be better to directly check for <sys/random.h> instead of checking the various glibc versions (which will fail anyway for anyone _not_ using glibc).
Can you update ccan/tools/configurator/configurator.c to check for sys/random.h, about time that it does something useful.
And we'll need a similar test for meson ...

Reply:
nice suggestion added both, ccan & meson :)